src/rust/engine/src/nodes.rs

```diff
@@ -1225,20 +1231,34 @@ impl Task {
             .await?
         };
 
-        let func = self.task.func.clone();
+        let args = self.args;
 
         let (mut result_val, mut result_type) =
             maybe_side_effecting(self.task.side_effecting, &self.side_effected, async move {
                 Python::with_gil(|py| {
-                    let func_val = func.0.to_value();
-                    let func = (*func_val).as_ref(py);
-                    externs::call_function(func, &deps)
-                        .map(|res| {
-                            let type_id = TypeId::new(res.get_type());
-                            let val = Value::new(res.into_py(py));
-                            (val, type_id)
-                        })
-                        .map_err(Failure::from)
+                    let func = (*self.task.func.0.value).as_ref(py);
+
+                    // If there are no explicit positional arguments, apply any computed arguments as
```

Comment:
Why not use kwargs in both cases?

Reply:
Because using `kwargs` involves constructing a dict, which afaik is a performance sink from a Python-calling-convention perspective.